File: Programming/ecobike_rental/lib/data/model/rental.dart
import 'package:equatable/equatable.dart';

import '../db/database.dart';

class Rental extends Equatable implements Model {
  Rental({this.id, this.rateContent, this.rateNumber});

  Rental.empty();

  Rental.fromMap(Map<String, dynamic> map) {
    id = map[Rental.key];
    renterId = map["renter_id"];
    bikeId = map["bike_id"];
    cardId = map["card_id"];
    rateContent = map["rate_content"];
    rateNumber = map["rate_number"];
  }

  static String tableName = "Rentals";
  static String key = "rental_id";

  int id;
  int renterId;
  int bikeId;
  int cardId;
  String rateContent;
  double rateNumber;

  @override
  List<Object> get props => [id];

  @override
  Map<String, dynamic> toMap() {
    return {
      Rental.key: id,
      'renter_id': renterId,
      'bike_id': bikeId,
      'card_id': cardId,
      'rate_content': rateContent,
      'rate_number': rateNumber
    };
  }

  @override
  String getTableName() {
    return Rental.tableName;
  }
}

File: TKXDPM.Server/Model/Rental.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TKXDPM_API.Model
{
    public class Rental
    {
        [Key] public int RentalId { get; set; }
        public int BikeId { get; set; }
        public Bike Bike { get; set; }
        public int CardId { get; set; }
        public Card Card { get; set; }
        public int RenterId { get; set; }
        public Renter Renter { get; set; }
        [Column(TypeName = "varchar(255)")]
        public string RateContent { get; set; }
        public int RateNumber { get; set; }
        
        public static List<Rental> GetSeederData()
        {
            return new List<Rental>
            {
                new Rental()
                {
                    RentalId = 100001,BikeId = 100001,Bike = new Bike(),CardId = 100001,Card = new Card(),
                    RenterId = 100001, Renter = new Renter(),RateContent = "Mọi thứ đều tốt.",RateNumber = 5
                },
                new Rental()
                {
                    RentalId = 100002,BikeId = 100002,Bike = new Bike(),CardId = 100002,Card = new Card(),
                    RenterId = 100002, Renter = new Renter(),RateContent = "Tuyệt vời.",RateNumber = 5
                },
                new Rental()
                {
                    RentalId = 100003,BikeId = 200001,Bike = new Bike(),CardId = 100003,Card = new Card(),
                    RenterId = 100003, Renter = new Renter(),RateContent = "Mọi thứ đều tốt.",RateNumber = 5
                },
                new Rental()
                {
                    RentalId = 100004,BikeId = 200002,Bike = new Bike(),CardId = 100001,Card = new Card(),
                    RenterId = 100001, Renter = new Renter(),RateContent = "Xe đi rất thoải mái.",RateNumber = 4
                },
                new Rental()
                {
                    RentalId = 100005,BikeId = 300001,Bike = new Bike(),CardId = 100001,Card = new Card(),
                    RenterId = 100001, Renter = new Renter(),RateContent = "Mọi thứ đều tốt.",RateNumber = 5
                },
                new Rental()
                {
                    RentalId = 100006,BikeId = 100004,Bike = new Bike(),CardId = 100001,Card = new Card(),
                    RenterId = 100001, Renter = new Renter(),RateContent = "Trải nghiệm thú vị.",RateNumber = 5
                },
                
            };
        }
        
    }

    public class RentalResponse
    {
        public RentalResponse()
        {
            Renter = new Renter();
            Bike = new BikeResponse();
            RateContent = "Good";
            RateNumber = 5;
        }

        public Renter Renter { get; set; }
        public BikeResponse Bike { get; set; }
        public string RateContent { get; set; }
        public int RateNumber { get; set; }
    }
}